### PR TITLE
Adding short README files for each XIOS example

### PR DESCRIPTION
**Pull request by svadams**
I have added READMEs based upon my understanding of what each toy does. We could add more information - possibly more details of what the known failure tests are.

### CODE REVIEW

File: xios_examples/read_domain_resample/README [hunk 1]
```diff
@@ -0,0 +1,4 @@
+read_domain_resample
+--------------------
+
+This example reads in a sample NetCDF file of arbitrary data on a 5 x 5 rectilinear (structured) horizontal domain and resamples to a 4 x 4 rectilinear horizontal domain using bilinear interpolation. The original and resampled data as well as a diff between them are written to one output NetCDF file. The unit tests include two known failure test cases where the expected result is not produced.
```
Comment (mo-marqh) [inline xios_examples/read_domain_resample/README:4]: this example set includes results output to a 4x4 array and a 2x2 array

as we may want to add more cases we may want to simply not include this level of resolution detail in the readme, instead writing something like

`These examples read in arbitrary data from a netCDF file and resample to a different horizontal domain as defined within the input netCDF file using bilinear interpolation`

for the first sentence?

Comment (svadams) [reply xios_examples/read_domain_resample/README:4]: Thank you @mo-marqh - I think that's a good idea !

**Comment by mo-marqh** [comment]
hi @svadams 

i had some problems with the XIOS3 automated testing, which was failing on this proposal when i added a minor wording revision as discussed.

i tried and failed to fix this, so reimplemented your commit on #8 which supersedes this PR.
#8 is merged so this is no superseded, and all the README content adopted
many thanks, mark

**Comment by svadams** [comment]
OK thank you for letting me know @mo-marqh! I will delete the branch.

**Pull request closed without merging**